game/scripts/vscripts/Sections/Chapter5/SectionRoshan.ts

```diff
@@ -104,25 +104,25 @@ function onStart(complete: () => void) {
                 goalDefeatRoshan.start()
             }),
             tg.playGlobalSound(roshanMusic),
-            tg.completeOnCheck(() => {
-                return roshan.IsAttacking()
-            }, 0.5),
+            tg.completeOnCheck(() => !unitIsValidAndAlive(roshan) || roshan.IsAttacking(), 0.5),
             shared.spawnFriendlyHeroes(Vector(-2000, 1550, 0)),
             tg.wait(1),
             tg.immediate(context => {
-                for (const friendlyHero of friendlyHeroesInfo) {
-                    const hero: CDOTA_BaseNPC_Hero = context[friendlyHero.name]
-                    hero.SetForceAttackTarget(roshan)
+                if (unitIsValidAndAlive(roshan)) {
+                    for (const friendlyHero of friendlyHeroesInfo) {
+                        const hero: CDOTA_BaseNPC_Hero = context[friendlyHero.name]
+                        if (unitIsValidAndAlive(hero)) {
+                            hero.SetForceAttackTarget(roshan)
+                        }
+                    }
                 }
             }),
             tg.setCameraTarget(context => context[friendlyHeroesInfo[0].name]),
             tg.wait(1),
             tg.setCameraTarget(undefined),
             tg.immediate(_ => centerCameraOnHero()),
             tg.audioDialog(LocalizationKey.Script_5_Roshan_6, LocalizationKey.Script_5_Roshan_6, ctx => ctx[CustomNpcKeys.SunsFanMudGolem]),
-            tg.completeOnCheck(() => {
-                return !roshan.IsAlive()
-            }, 2),
+            tg.completeOnCheck(() => !unitIsValidAndAlive(roshan), 2),
```

Comment:
I know this was like that already, but can't the timer to be shorter, to potentially not have to wait almost 2 seconds to get the system to notice that Roshan's dead?

We can make it super short, 0.25 or something.

Reply:
Fixing this, also a couple others in the same section with 2s instead of something lower